lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -968,6 +968,56 @@ static LogicalResult verifyConnectOp(ConnectOp connect) {
   return success();
 }
 
+static LogicalResult verifyPartialConnectOp(PartialConnectOp partialConnect) {
+  FIRRTLType destType = partialConnect.dest().getType().cast<FIRRTLType>();
+  FIRRTLType srcType = partialConnect.src().getType().cast<FIRRTLType>();
+
+  auto isPortOrInstancePort = [](Value a) -> bool {
+    auto op = a.getDefiningOp();
+    return !op || isa<InstanceOp>(op);
+  };
+
+  // If the source or destination is a port or instance port, then an optional
+  // outer flip, indicating the direction (input or output), should be stripped
+  // for type checking.
+  if (isPortOrInstancePort(partialConnect.dest()))
+    if (auto destTypeFlip = destType.dyn_cast<FlipType>())
+      destType = destTypeFlip.getElementType();
+  if (isPortOrInstancePort(partialConnect.src()))
+    if (auto srcTypeFlip = srcType.dyn_cast<FlipType>())
+      srcType = srcTypeFlip.getElementType();
```

Comment:
This might not be worth asking about, since input and output are going to become bitvectors on the op, but are block arguments and ports the only things that can have an outer flip type? Should we just strip the outer flip type without checking where it comes from?

Reply:
Good idea. Yes, this should work. The one other case (that this is actually missing) is that outer flips should be stripped for memory ports, too. Blindly stripping the flip is the simplest solution here.

Reply:
As this is now basically duplicating the flip stripping lambda in `areTypesWeaklyEquivalent` I added a commit that creates `std::pair<FIRRTLType, bool> FIRRTLType::stripFlip()`. This will live until the bitvector gets added.